lib/postgrex/utils.ex

```diff
@@ -52,10 +52,18 @@ defmodule Postgrex.Utils do
     |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
     |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
     |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
-    |> Keyword.put_new(:port, System.get_env("PGPORT"))
+    |> Keyword.put_new(:port, set_port(System.get_env("PGPORT")))
     |> Enum.reject(fn {_k, v} -> is_nil(v) end)
   end
 
+  defp set_port(port) when is_binary(port) do
```

Comment:
The `set_port` name is a bit confusing, good in test but not here.
Maybe `port_number` or `port_value`?